### PR TITLE
Topology: Allow pass of capture PGA and controls name from upper level

### PR DESCRIPTION
**Pull request by singalsu**
This patch allows to change Dmic0 and Dmic1 processing pipelines
    to e.g. passthrough, volume, or eq-iir-volume capture with simple
    -DDMICPROC=x and -DDMIC16KPROC=y definitions from topology
    CMakeLists.txt.
    
    Most of the changed m4 code lines are from change to avoid to hard
    code volume control and switch names for Dmic0 and Dmic1 in the
    pipeline. If they are not defined the normal name with PIPELINE_ID
    is used.
    
    A new pipeline macro pipe-passthrough-16khz.m4 is created for being
    able to set DMIC16KPROC=passthrough.
    
    In pipe-volume-switch-capture the PGA tokens are renamed with macros
    to contain pipeline ID to avoid conflict with other pipelines.
    
    Two example topologies are build to topology/development:
    sof-hda-generic-2ch-volume and sof-hda-generic-2ch-passthrough.
    
    Signed-off-by: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>

### CODE REVIEW

**Comment by singalsu** [comment]
There's a simple way to preserve the control names "PIPELINE_ID Master Capture Volume" if not defined. I'll update soon!

File: tools/topology/development/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,8 @@ set(TPLGS_UP
 	"sof-cml-rt5682\;sof-cml-eq-iir-rt5682\;-DPLATFORM=cml\;-DHSEARPROC=eq-iir-volume\;-DDMICPROC=eq-iir-volume\;-DDMIC16KPROC=eq-iir-volume"
 	"sof-glk-da7219\;sof-glk-eq-da7219\;-DDMICPROC=eq-iir-volume"
 	"sof-hda-generic\;sof-hda-generic-tdfb_50mm-2ch\;-DCHANNELS=2\;-DHSPROC=volume\;-DDMIC16KPROC=tdfb-eq-iir-volume\;-DDMIC16KPROC_FILTER1=tdfb/coef_line2_50mm_pm10deg_16khz.m4\;-DDMICPROC=tdfb-eq-iir-volume\;-DDMICPROC_FILTER1=tdfb/coef_line2_50mm_pm10deg_48khz.m4\;-DDMICPROC_FILTER2=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER2=eq_iir_coef_highpass_40hz_20db_16khz.m4"
+	"sof-hda-generic\;sof-hda-generic-2ch-volume\;-DCHANNELS=2\;-DHSPROC=volume\;-DDMICPROC=volume\;-DDMIC16KPROC=volume"
```
[the remark this answers is not in the document]
Comment (singalsu) [reply tools/topology/development/CMakeLists.txt]: Done

**Comment by singalsu** [comment]
@plbossart @juimonen @libinyang Do you have recommendations for PCM names. They do change with every pipeline if the processing is changed "Passthrough Capture", "Highpass Capture", "TDFB Capture", etc.

**Comment by lgirdwood** [comment]
@plbossart ping any preference on the PCM naming. Can we do any alignment with the soundwire PCM list ?

**Comment by plbossart** [comment]
> @plbossart ping any preference on the PCM naming. Can we do any alignment with the soundwire PCM list ?

why do we want to change the PCM names? It took us multiple rounds already to converge to the existing solution (DMIC, DMIC16kHz, BufferedMic)?

**Comment by lgirdwood** [comment]
> > @plbossart ping any preference on the PCM naming. Can we do any alignment with the soundwire PCM list ?
> 
> why do we want to change the PCM names? It took us multiple rounds already to converge to the existing solution (DMIC, DMIC16kHz, BufferedMic)?

@plbossart just checking, I knew you had some opinions on naming scheme for new stuff.
@singalsu can you give us an example of how this changes the PCM names i.e. from and to

**Comment by singalsu** [comment]
I was wondering if e.g. endpoint name would be better than current fairly random ones. There's even some copy-paste mistakes with names. Some names claim pass-through even if the pipeline does some processing.

But if there are no issues I'm happy to leave non-mandatory cleanup to topology 2.0.

**Comment by singalsu** [comment]
@juimonen Ping, did this PR fix for you the issue you had in replacing capture pipeline with IIR with plain volume or passthrough capture?

**Comment by paulstelian97** [comment]
Please resubmit with "main" as PR base branch.

**Pull request closed without merging**